### PR TITLE
wxwidgets: fix assertion using hide in destroy

### PR DESCRIPTION
**Pull request by ylwghst**
###### Motivation for this change

I have been experiencing errors while using FileZilla due to the bug in wxwidgets library.
Annoying pop up with error message is appearing every time FileZilla is started.

> 
>  wxTopLevelWindowBase::Destroy calls Hide on itself if there is at least one other top level window.
> Unfortunately calling Hide on a window that has not yet been created causes an assertion on wxGTK.
> It is very easy to reproduce with this snippet you could for example add to the minimal sample in 
> 
> MyFrame::OnAbout:
> (new wxDialog)->Destroy();
> 
> Both WX_3_0_BRANCH and master are affected by this issue.
> 
> 

The error messages is here: https://gist.github.com/anonymous/38f27cbdc43c96a9c7fe063c3a10a08c

The bug is tracked here: https://trac.wxwidgets.org/ticket/17942

Patch source available here: https://trac.wxwidgets.org/attachment/ticket/17942/fix_assertion_using_hide_in_destroy.diff

> 
> I have attached a patch for this issue. With it, Hide is only called if the window is actually shown.
> 
> The attached patch has been created against WX_3_0_BRANCH but also applies cleanly against master.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [+] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [+] Tested execution of all binary files (usually in `./result/bin/`)
- [+] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/wxwidgets/2.8/default.nix [hunk 1]
```diff
@@ -22,6 +22,10 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ pkgconfig ];
 
+  patches = [
+   ./patches/fix_assertion_using_hide_in_destroy.diff
```
Comment (jtojnar) [inline pkgs/development/libraries/wxwidgets/2.8/default.nix]: The patch should not be placed into the repository but fetched using `fetchurl`.

**Review by matthewbauer — APPROVED** [review]

**Comment by disassembler** [comment]
Looks good to me. @vcunat you want this rebased on staging with the 100-500 linux rebuilds it affects?

**Comment by vcunat** [comment]
100&ndash;500 seems generally OK for master.  Nitpick: the patches seem non-generated, so there's no real advantage of using `fetchpatch` instead of `fetchurl`, but it doesn't really matter.

**Comment by vcunat** [comment]
There were multiple issues, so I rewrote it.  The URL was to a patch *rendered* as HTML, the patch won't apply to 2.8, and the hash didn't seem to match.  I left 2.8 unpatched (it seems rarely used anyway).

**Pull request closed without merging**